src/callbacks/callback_save_model.cpp

```diff
@@ -133,4 +135,42 @@ bool lbann_callback_save_model::save_model_weights(model *m) {
   return true;
 }
 
+bool lbann_callback_save_model::load_model_weights(std::string ckpt_dir, model * m) {
+  std::vector<std::string> weight_list = std::vector<std::string>();
+  int epochLast = -1;
+  int stepLast = -1;
+  std::string latest;
+  latest = get_last_shared_checkpoint_filename(m, ckpt_dir);
+
+  // get last epoch and step saved.
+  int success = read_latest(latest, &epochLast, &stepLast);
+  if(!success) {
+    return false;
+  }
+  lbann_comm *comm = m->get_comm();
+  if(comm->am_model_master()) {
+    latest = get_shared_checkpoint_dirname(m, ckpt_dir, epochLast, stepLast);
+    std::cout << "Loading model weights from " << latest << std::endl;
+  }
+  comm->model_broadcast(0, &(latest[0]), sizeof(latest), El::SyncInfo<El::Device::CPU>{});
+
+  DIR *weight_dir;
+  struct dirent *weight_file;
+  if((weight_dir = opendir(latest.c_str())) == NULL)
```

Comment:
This is too clever. It's more readable if you break it up into two lines. We should also prefer `nullptr` to `NULL`.

Reply:
Fixed.